### PR TITLE
Provide in api/status the git commit metadata (SHA/branch/tag) used to make a Vercel deployment

### PR DESCRIPTION
**Pull request by Demmonius**
https://app.asana.com/0/1136912812619441/1199380329316502

Make GIT_COMMIT_REF and GIT_COMMIT_SHA available as build environment variables.
Useful if you need to know what git commit hash/branch/tag was used to make a deployment.

### CODE REVIEW

File: .github/workflows/deploy-vercel-staging.yml [hunk 1]
```diff
@@ -67,7 +69,10 @@ jobs:
 
           # Deploy the customer on Vercel using the customer ref
           # Store the output in a variable so we can extract metadata from it
-          VERCEL_DEPLOYMENT_OUTPUT=`CUSTOMER_REF=${CUSTOMER_REF_TO_DEPLOY} yarn deploy:customer --token $VERCEL_TOKEN`
+          echo "GIT_COMMIT_SHA" $GIT_COMMIT_SHA
+          echo "github.sha" ${{ github.sha }}
+          echo "VERCEL_GITHUB_COMMIT_SHA" $VERCEL_GITHUB_COMMIT_SHA
```
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-staging.yml]: this displays properly

File: next.config.js [hunk 2]
```diff
@@ -14,6 +15,13 @@ const noRedirectBlacklistedPaths = ['_next', 'api']; // Paths that mustn't have
 const publicBasePaths = ['robots', 'static', 'favicon.ico']; // All items (folders, files) under /public directory should be added there, to avoid redirection when an asset isn't found
 const noRedirectBasePaths = [...supportedLocales, ...publicBasePaths, ...noRedirectBlacklistedPaths]; // Will disable url rewrite for those items (should contain all supported languages and all public base paths)
 const date = new Date();
+const commitInfo = gitCommitInfo();
+const GIT_COMMIT_SHA = process.env.GIT_COMMIT_SHA || (commitInfo && commitInfo.commit);
```
Comment (Vadorequest) [inline next.config.js]: None of those logs print anything, sometimes it's undefined, sometimes empty

File: src/pages/api/status.ts [hunk 3]
```diff
@@ -43,6 +43,8 @@ export const status = async (req: NextApiRequest, res: NextApiResponse): Promise
       environment: process.env.NODE_ENV,
       preset: process.env.NEXT_PUBLIC_NRN_PRESET,
       customerRef: process.env.NEXT_PUBLIC_CUSTOMER_REF,
+      GIT_COMMIT_SHA: process.env.GIT_COMMIT_SHA,
```
Comment (Vadorequest) [inline src/pages/api/status.ts:46]: Those don't print anything at https://nrn-v2-mst-aptd-at-lcz-sty-c1-implement-s.vercel.app/api/status

File: .github/workflows/deploy-vercel-staging.yml (hunk 1 above)
Comment (Vadorequest) [reply .github/workflows/deploy-vercel-staging.yml]: image

https://github.com/UnlyEd/next-right-now/runs/1566025695

File: .github/workflows/deploy-vercel-staging.yml [hunk 4]
```diff
@@ -136,6 +141,7 @@ jobs:
         env:
           VERCEL_TOKEN: ${{ secrets.VERCEL_TOKEN }} # Passing github's secret to the worker
           CURRENT_BRANCH: ${{ github.ref }} # Passing current branch to the worker
```
Comment (Vadorequest) [inline .github/workflows/deploy-vercel-staging.yml]: GIT_COMMIT_REF

Comment (Vadorequest) [reply .github/workflows/deploy-vercel-staging.yml]: + forward it too to display it in /status api

Comment (Vadorequest) [reply .github/workflows/deploy-vercel-staging.yml]: + adapt comment

File: next.config.js [hunk 5]
```diff
@@ -74,6 +74,7 @@ module.exports = withBundleAnalyzer(withSourceMaps({
     NEXT_PUBLIC_APP_VERSION: packageJson.version,
     NEXT_PUBLIC_APP_NAME_VERSION: `${packageJson.name}-${packageJson.version}`,
     UNLY_SIMPLE_LOGGER_ENV: process.env.NEXT_PUBLIC_APP_STAGE, // Used by @unly/utils-simple-logger - Fix missing staging logs because otherwise it believes we're in production
+    GIT_COMMIT_SHA: process.env.GIT_COMMIT_SHA || (commitInfo && commitInfo.commit) //
```
Comment (Vadorequest) [inline next.config.js]: TODO comment

File: package.json [hunk 6]
```diff
@@ -16,18 +16,19 @@
     "deploy:all": "yarn deploy:customer1 && yarn deploy:customer2",
     "deploy:all:production": "yarn deploy:customer1:production && yarn deploy:customer2:production",
     "deploy:all:all": "yarn deploy:all && yarn deploy:all:production",
-    "deploy:customer": "yarn vercel:cleanup && vercel --confirm --local-config=vercel.${CUSTOMER_REF}.staging.json --debug",
-    "deploy:customer:production:simple": "yarn vercel:cleanup && vercel --confirm --local-config=vercel.${CUSTOMER_REF}.production.json --prod --debug",
+    "deploy:customer": "yarn vercel:cleanup && yarn vercel:deploy --local-config=vercel.${CUSTOMER_REF}.staging.json",
+    "deploy:customer:production:simple": "yarn vercel:cleanup && yarn vercel:deploy --local-config=vercel.${CUSTOMER_REF}.production.json --prod",
```
Comment (Vadorequest) [inline package.json:20]: Why did you remove `confirm` option?

File: vercel.customer1.staging.json [hunk 7]
```diff
@@ -15,7 +15,7 @@
       "LOCIZE_API_KEY": "@nrn-locize-api-key-staging",
       "SENTRY_DSN": "@nrn-sentry-dsn",
       "GITHUB_DISPATCH_TOKEN": "@nrn-github-dispatch-token"
-    }
+}
```
Comment (Vadorequest) [inline vercel.customer1.staging.json:18]: regression

**Review by Vadorequest — DISMISSED** [review]

File: package.json (hunk 6 above)
Comment (Vadorequest) [reply package.json:20]: ah, because it's included in vercel:deploy, nevermind

File: package.json [hunk 8]
```diff
@@ -16,18 +16,19 @@
...
+    "deploy:customer1:production:simple": "yarn vercel:cleanup && yarn vercel:deploy --local-config=vercel.customer1.production.json --prod",
     "deploy:customer2:all": "yarn deploy:customer2 && yarn deploy:customer2:production",
     "deploy:customer2": "yarn vercel:cleanup && vercel --confirm --local-config=vercel.customer2.staging.json --debug",
     "deploy:customer2:production": "yarn deploy:customer2:production:simple && yarn e2e:customer2:production",
-    "deploy:customer2:production:simple": "yarn vercel:cleanup && vercel --confirm --local-config=vercel.customer2.production.json --prod --debug",
+    "deploy:customer2:production:simple": "yarn vercel:cleanup && yarn vercel:deploy --local-config=vercel.customer2.production.json --prod",
     "deploy:fake": "git commit --allow-empty -m \"Fake empty commit (force CI trigger)\"",
     "vercel:cleanup": "npx del-cli .vercel/ && npx del-cli .vercel/",
+    "vercel:deploy": "vercel -e GIT_COMMIT_SHA=${GIT_COMMIT_SHA:?You need to provide a commit SHA in order to build} --confirm --debug",
```
Comment (Vadorequest) [inline package.json]: Add GIT_COMMIT_REF too

File: package.json [hunk 9]
```diff
@@ -16,18 +16,19 @@
     "deploy:all": "yarn deploy:customer1 && yarn deploy:customer2",
     "deploy:all:production": "yarn deploy:customer1:production && yarn deploy:customer2:production",
     "deploy:all:all": "yarn deploy:all && yarn deploy:all:production",
-    "deploy:customer": "yarn vercel:cleanup && vercel --confirm --local-config=vercel.${CUSTOMER_REF}.staging.json --debug",
-    "deploy:customer:production:simple": "yarn vercel:cleanup && vercel --confirm --local-config=vercel.${CUSTOMER_REF}.production.json --prod --debug",
+    "deploy:ci:gha": "yarn vercel:cleanup && echo 'XXX - GIT_COMMIT_REF: ' $GIT_COMMIT_REF && echo 'GIT_COMMIT_SHA: ' $GIT_COMMIT_SHA && yarn vercel:deploy --local-config=vercel.${CUSTOMER_REF}.staging.json",
```
Comment (Vadorequest) [inline package.json:19]: @Demmonius I tried to echo here but don't see those printed on any log. Is that normal?

Comment (Demmonius) [reply package.json:19]: Not really, I'm going to have a look !
